packages/usa-input-mask/src/index.js

```diff
@@ -0,0 +1,162 @@
+const Sanitizer = require("../../uswds-core/src/js/utils/sanitizer"); // eslint-disable-line no-unused-vars
+
+const MASKED = "masked";
+const MASKED_CLASS = `.${MASKED}`;
+const SHELL = "shell";
+const PLACEHOLDER = "placeholder";
+
+function ready(fn) {
+  document.addEventListener("DOMContentLoaded", fn, false);
+}
+
+ready(() => {
+  const masking = {
+    // User defined Values
+    maskedInputs: document.getElementsByClassName(MASKED),
+    maskedNumber: "XdDmMyY9",
+    maskedLetter: "_",
+
+    init() {
+      masking.setUpMasks(masking.maskedInputs);
+      masking.maskedInputs = document.querySelectorAll(MASKED_CLASS); // Repopulating. Needed b/c static node list was created above.
+      masking.activateMasking(masking.maskedInputs);
+    },
+
+    setUpMasks(inputs) {
+      let i;
+      const l = inputs.length;
+
+      for (i = 0; i < l; i += 1) {
+        masking.createShell(inputs[i]);
+      }
+    },
+
+    // replaces each masked input with a shell containing the input and it's mask.
+    createShell(input) {
+      let text = "";
+      const placeholder = input.getAttribute(`${PLACEHOLDER}`);
+
+      if (
+        placeholder !== null &&
+        placeholder.length > 0 &&
+        placeholder !== " "
+      ) {
+        input.setAttribute("maxlength", placeholder.length);
+        input.setAttribute("data-placeholder", placeholder);
+        input.removeAttribute(`${PLACEHOLDER}`);
+      } else {
+        return;
+      }
+
+      // This section still needs some rework
+      text = `<span class="${SHELL}"><span aria-hidden="true" id="${input.id}Mask"><i></i>${placeholder}</span>${input.outerHTML}</span>`;
+
+      input.outerHTML = text; // eslint-disable-line no-param-reassign, no-unsanitized/property
+    },
+
+    setValueOfMask(e) {
+      const { value } = e.target;
+      const placeholder = e.target.getAttribute("data-placeholder");
+      const placeholderVal = `${placeholder.substr(value.length)}`;
+
+      const theIEl = document.createElement("i");
+      theIEl.textContent = value;
+      return [theIEl, placeholderVal];
+    },
+
+    // add event listeners
+    activateMasking(inputs) {
+      let i;
+      let l;
+
+      for (i = 0, l = inputs.length; i < l; i += 1) {
+        masking.maskedInputs[i].addEventListener(
+          "keyup",
+          (e) => {
+            masking.handleValueChange(e);
+          },
+          false
+        );
+      }
+    },
+
+    handleValueChange(e) {
+      const id = e.target.getAttribute("id");
+
+      switch (
+        e.keyCode // allows navigating thru input
+      ) {
+        case 20: // caplocks
+        case 17: // control
+        case 18: // option
+        case 16: // shift
+        case 37: // arrow keys
+        case 38:
+        case 39:
+        case 40:
+        case 9: // tab (let blur handle tab)
+          return;
+        // no default
+      }
```

Comment:
Alternatively, there's a keymap functino you could use. See date picker example:

https://github.com/uswds/uswds/blob/cda1459efd1c028fce62c960eafc2653993654f9/packages/usa-date-picker/src/index.js#L2127-L2143

Reply:
I removed the key mapping here, it might have been overkill - I tested with VO screen reader + keyboard and I didn't notice a difference. We can add it back in if it's necessary.